flask_monitoringdashboard/frontend/js/controllers/OverviewController.js

```diff
@@ -28,23 +30,57 @@ export function OverviewController($scope, $http, $location, menuService, endpoi
         $scope.table = response.data;
     });
 
-    function getItemsForPage(pageNumber) {
-        const start = Number(pageNumber) * $scope.pageSize;
-        const end = (Number(pageNumber) + 1) * Number($scope.pageSize);
+    function sortOrder(a, b){
+      return a[$scope.sortBy] > b[$scope.sortBy] || b[$scope.sortBy] === null;
+    }
+    
+    function sortItems(items){
+      return $scope.getOrAddSortingOrder($scope.sortBy) ? items.sort(sortOrder) : items.sort((a, b) => sortOrder(b, a))
+    }
 
+    function getItemsForPage(pageNumber) {
+        const start = pageNumber * Number($scope.pageSize);
+        const end = (pageNumber + 1) * Number($scope.pageSize);
+  
         let items = $scope.table
-            .filter(item => item.name.includes($scope.searchQuery))
+            .filter(item => item.name.includes($scope.searchQuery));
+
         if ($scope.slectedBlueprint) {
-            items = items.filter(item => item.blueprint===$scope.slectedBlueprint)
+            items = items.filter(item => item.blueprint===$scope.slectedBlueprint);
         }
+    
+        return sortItems(items).slice(start, end);
+    }
 
-        return items.slice(start, end);
+    $scope.getOrAddSortingOrder = function (column) {
+        let orderBy = $scope.sortingOrder[column];
+        if (orderBy !== undefined) {
+            return orderBy;
+        }
+        $scope.sortingOrder[column] = true;
```

Comment:
it's hard to read this. could it be that the name of the `sortingOrder` variable should be something else? e.g. `isSortedBy`? after all it seems we're setting it as a boolean, so it would be natural to hame it as a predicate. Or is it tracking whether it's ascending or descending? In that case it would be easier to follow if two constants would be introduced to track that, e.g. `sortingOrder = ASCENDING` vs `DESCENDING`. Or simply name the variable `isSortedAscending`?